Flask test_client json not serialized using client.post

[see recent fail](https://github.com/andromia/cvrp-app-rpc/runs/815020700?check_suite_focus=true)
```
 ============================= test session starts ==============================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/runner/work/cvrp-app-rpc/cvrp-app-rpc, inifile: pytest.ini
collected 1 item

tests/test_app.py::test_main_procedure FAILED                            [100%]

=================================== FAILURES ===================================
_____________________________ test_main_procedure ______________________________

client = <FlaskClient <Flask 'app'>>

    def test_main_procedure(client):
        input_data = json.dumps(dict())
        endpoint = '/api/%s/procedure' % __version__
        logging.debug('endpoint: %s.' % endpoint)
    
>       response = client.post(endpoint, data=input_data)

tests/test_app.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
```

Local Environment: [Vagrant VM](https://github.com/andromia/cvrp-app-rpc/blob/master/Vagrantfile) ✔️

--- a/app/api/v0_1/main.py
+++ b/app/api/v0_1/main.py
@@ -1,5 +1,5 @@
 from . import bp
-
+from json import loads
 from flask import request, jsonify
 
 
@@ -21,6 +21,8 @@ def main_procedure():
         "vehicles": [] # optional
     }
     """
-    data = request.data
+
+    # request.data is a bytestring
+    data = loads(request.data)
 
     return jsonify(data)

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -19,8 +19,10 @@ def client():
 
 def test_main_procedure(client):
     input_data = json.dumps(common.get_vrp_data())
-    endpoint = '/api/%s/procedure' % __version__
-    logging.debug('endpoint: %s.' % endpoint)
+    logging.debug(f'input data : {input_data}')
+
+    endpoint = f'/api/{__version__}/procedure'
+    logging.debug(f'endpoint: {endpoint}')
 
     response = client.post(endpoint, data=input_data)
     output = json.loads(response.get_data())

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -24,7 +24,7 @@ def test_main_procedure(client):
     endpoint = f'/api/{__version__}/procedure'
     logging.debug(f'endpoint: {endpoint}')
 
-    response = client.post(endpoint, data=input_data)
+    response = client.post(endpoint, json=input_data)
     output = json.loads(response.get_data())
 
     assert output == input_data

